tests/quantization/autoawq/test_awq.py

```diff
@@ -382,3 +386,24 @@ def test_generation_custom_model(self):
 
         outputs = model.generate(**inputs, max_new_tokens=12)
         self.assertEqual(tokenizer.decode(outputs[0], skip_special_tokens=True), self.EXPECTED_GENERATION_CUSTOM_MODEL)
+
+    @require_torch_multi_gpu
+    def test_generation_mixtral_fused(self):
+        """
+        Text generation test for Mixtral + AWQ + fused
+        """
```

Comment:
Do we need a model specific test here? We don't want to have to add tests for every model we cover. It would be better to have tests which cover different functional properties e.g. A, B, C. Then if any model uses A & C we know it works

Reply:
This test should be generalizable to all `mixtral` models as the many thing to make sure that it works is on the interaction between `modules_to_not_convert` and fused modules for `mixtral` ! 
I can also do a smaller test with a tiny model - in addition to this one, if we know that the tiny model is correctly loaded then other models should be correctly loaded as well - wdyt? I would say in general this test is also good to have as the underlying things that it tests are 

1- correct conversion of mixtral to mixtral fused modules (with `modules_to_not_convert` being properly set)
2- Generation correctness for mixtral + fused modules
3- Batched generation correctness for mixtral fused modules

Reply:
Yes please - let's add a more general test for a tiny model to make sure the code works generally: we don't want to overfit to specifics of mixtral but also want to make sure mixtral works.

Reply:
perfect, will do !

Reply:
Done!

Reply:
[3e1bc0b](https://github.com/huggingface/transformers/pull/28240/commits/3e1bc0b21894efd6f4ed1474b06ad29e7d7dfcc8)